app/models/subnet.rb

```diff
@@ -1,17 +1,37 @@
 require 'ipaddr'
+
 class Subnet < ActiveRecord::Base
+  def self.default_sti_class
+    'Subnet::Ipv4' if self.name == 'Subnet'
+  end
+
+  IP_FIELDS = [:network, :mask, :gateway, :dns_primary, :dns_secondary, :from, :to]
```

Comment:
It looks like it would make sense to have an `Ip` class that hold all the validations, changes, etc...